content/docs/agentgateway/mcp/dynamic-mcp.md

```diff
@@ -0,0 +1,196 @@
+---
+title: Dynamic MCP
+weight: 20
+---
+
+Route to a dynamic Model Context Protocol (MCP) server by using a label selector. For more information, see the [About MCP](../#about) topic.
+
+{{< callout type="warning" >}}
+Note that only streamable HTTP is currently supported for label selectors. If you have SSE, use a [static MCP Backend](../static-mcp/).
+{{< /callout >}}
+
+## Before you begin
+
+{{< reuse "docs/snippets/prereq-agw.md" >}}
+
+## Step 1: Deploy an MCP server {#mcp-server}
+
+Deploy an MCP server that you want agentgateway to proxy traffic to. The following example sets up an MCP server that provides various utility tools.
+
+1. Create an MCP server (`mcp-server`) that provides various utility tools. Notice that the Service uses the `appProtocol: kgateway.dev/mcp` setting. This way, kgateway configures the agentgateway proxy to use MCP for the Backend that you create in the next step.
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     name: mcp-server-everything
+     labels:
+       app: mcp-server-everything
+   spec:
+     replicas: 1
+     selector:
+       matchLabels:
+         app: mcp-server-everything
+     template:
+       metadata:
+         labels:
+           app: mcp-server-everything
+       spec:
+         containers:
+           - name: mcp-server-everything
+             image: node:20-alpine
+             command: ["npx"]
+             args: ["-y", "@modelcontextprotocol/server-everything", "streamableHttp"]
+             ports:
+               - containerPort: 3001
+   ---
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: mcp-server-everything
+     labels:
+       app: mcp-server-everything
+   spec:
+     selector:
+       app: mcp-server-everything
+     ports:
+       - protocol: TCP
+         port: 3001
+         targetPort: 3001
+         appProtocol: kgateway.dev/mcp
+     type: ClusterIP
+   EOF
+   ```
+
+2. Create a Backend that uses label selectors.
+
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.kgateway.dev/v1alpha1
+   kind: Backend
+   metadata:
+     name: mcp-backend
+   spec:
+     type: MCP
+     mcp:
+       name: mcp-virtual-server
+       targets:
+         - selectors:
+             serviceSelector:
+               matchLabels:
+                 app: mcp-server-everything
+   EOF
```

Comment:
do we know why this is needed? and why it can't just use an HTTPRoute and route to the service directly?